src/storage.rs

```diff
@@ -36,16 +70,99 @@ impl Storage {
     /// # Description
     /// Closes all file descriptors
     /// # Examples
+
+    // @TODO specify more useful error type
     pub fn close(&mut self) -> Result<(), ()> {
         // @TODO implement
         Ok(())
     }
+
+    /// # Description
+    /// Blobs count contains closed blobs and one opened, if is some
+    /// # Examples
+    /// ```
+    /// use pearl::Builder;
+    ///
+    /// let mut storage = Builder::new().work_dir("/tmp/pearl/").build();
+    /// storage.init();
+    /// assert_eq!(storage.blobs_count(), 1);
+    /// ```
+    pub fn blobs_count(&self) -> usize {
+        self.blobs.len() + if self.opened_blob.is_some() { 1 } else { 0 }
+    }
+}
+
+impl Storage {
+    fn prepare_work_dir(&self) -> io::Result<()> {
+        let wd = Path::new(&self.config.work_dir);
+        if let Err(e) = fs::read_dir(wd) {
+            match e.kind() {
+                ErrorKind::NotFound => {
+                    error!("\"{}\" not found", self.config.work_dir);
+                }
+                _ => return Err(e),
+            }
+        }
+        debug!(
+            "create work dir recursively: {}",
+            wd.to_str().unwrap_or("failed to convert path to str")
+        );
+        fs::create_dir_all(wd)?;
+        Ok(())
+    }
+
+    // @TODO specify more useful error type
+    #[inline]
+    fn init_opened_blob(&mut self) -> Result<(), ()> {
+        self.opened_blob = Box::new(Some(Default::default()));
+        Ok(())
+    }
+
+    // @TODO specify more useful error type
+    fn init_new(&mut self) -> Result<(), ()> {
+        self.init_opened_blob()
+    }
+
+    // @TODO specify more useful error type
+    fn init_from_existing(&mut self, files: Vec<DirEntry>) -> Result<(), ()> {
+        self.blobs = files
+            .iter()
+            .filter_map(|entry| {
+                // @TODO implement more file validations
+                if entry.metadata().ok()?.is_file() {
+                    Some(Blob::from_file(entry.path()).ok()?)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        if let Some(blob_index) = self.search_for_opened_blob() {
+            self.opened_blob = Box::new(Some(self.blobs.remove(blob_index)));
```

Comment:
we can use `blobs` instead of opened_blobs (as we would have opened blobs and active one)

Reply:
We can't add data to closed blobs. But we can update metadata - ex: delete or update in future

Reply:
Yes, I mean that Storage already has two fields for 1 active and N closed blobs